src/sandstorm/grain.capnp

```diff
@@ -431,18 +587,20 @@ interface SessionContext {
   # _without_ actually passing that capability to the user, use `getSharedPermissions()` to detect
   # when the user's permissions change and implement it yourself.
 
-  offer @2 (cap :PowerboxCapability, requiredPermissions :PermissionSet) -> ();
+  offer @2 (cap :Capability, requiredPermissions :PermissionSet,
+            descriptor :PowerboxDescriptor, displayInfo :PowerboxDisplayInfo);
   # Offer a capability to the user.  A dialog box will ask the user what they want to do with it.
-  # Depending on the type of capability, different options may be provided.  All capabilities will
-  # offer the user the option to save the capability to their capability/grain store.  Other
-  # type-specific actions may be offered by the platform or by other applications.
+  # Depending on the type of capability (as indicated by `descriptor`), different options may be
+  # provided.  All capabilities will offer the user the option to save the capability to their
+  # capability/grain store.  Other type-specific actions may be offered by the platform or by other
+  # applications.
   #
   # For example, offering a UiView will give the user options like "open in new tab", "save to
   # grain list", and "share with another user".
   #
   # The capability is implicitly tied to the user as if via `tieToUser()`.
 
-  request @3 (query :PowerboxQuery) -> (cap :PowerboxCapability);
+  request @3 (query :List(PowerboxDescriptor)) -> (cap :Capability);
```

Comment:
If my query is a disjunction of descriptors, I'm going to want to which of them is satisfied by the result that I get back. Is there going to be a way for me to find that out without try/catch?

Maybe it makes more sense to change this signature so that `query` is not a list?